app/controllers/api/groups_controller.rb

```diff
@@ -142,11 +142,37 @@ def update_marks
 
     def set_mark_by_criteria(criteria, mark_to_change)
       if criteria.is_a?(FlexibleCriterion)
-        mark_to_change.mark = params[criteria.flexible_criterion_name].to_f
+        if params[criteria.flexible_criterion_name] == 'nil'
+          mark_to_change.mark = nil
+        else
+          mark_to_change.mark = params[criteria.flexible_criterion_name].to_f
+        end
       else
-        mark_to_change.mark = params[criteria.rubric_criterion_name]
+        if params[criteria.rubric_criterion_name] == 'nil'
+          mark_to_change.mark = nil
+        else
+          mark_to_change.mark = params[criteria.rubric_criterion_name]
+        end
       end
       mark_to_change.save
     end
+
+    # Return key:value pairs of group_name:group_id
+    def group_ids_by_name
+      reversed = Assignment.find_by_id(params[:assignment_id])
+                           .groups
+                           .inject({}) do |dict, group|
```

Comment:
This is okay (fold!) but I'd rather you use the `Hash` constructor, which takes a nested list `[ [key, val], ... ]` and returns a Hash.